app/dao/provider_details_dao.py

```diff
@@ -22,22 +19,11 @@ def get_provider_details_by_identifier(identifier):
 
 
 def get_alternative_sms_provider(identifier):
```

Comment:
yeah - we no longer want to fetch at this point, because we've got the separate fetch for provider details that includes the `with_for_update()` lock.